src/translations.cpp

```diff
@@ -52,6 +52,15 @@ static void reload_names()
 
 static bool sanity_checked_genders = false;
 
+// int version/generation that is incremented each time language is changed
+// used to invalidate translation cache
+static int current_language_version = INVALID_LANGUAGE_VERSION + 1;
+
+int details::get_current_language_version()
+{
+    return current_language_version;
+}
+
```

Comment:
Since we're using this versioned cache idea in a couple of places now it might be worth considering creating a common abstraction?  Not sure how easy that would be.

Reply:
Good idea, I'll think about it. However, I believe it's outside of the scope of this PR.